trustfall_core/src/graphql_query/query.rs

```diff
@@ -125,49 +125,59 @@ fn try_get_query_root(document: &ExecutableDocument) -> Result<&Positioned<Field
 
     match &document.operations {
         DocumentOperations::Multiple(mult) => {
-            return Err(ParseError::MultipleOperationsInDocument(mult.values().next().unwrap().pos))
+            return if mult.values().len() > 1 {
+                mult.values().next().expect("Could not iterate to first value in document.");
+                Err(ParseError::MultipleOperationsInDocument(
+                    mult.values()
+                        .next()
+                        .expect("Could not iterate to second value in document.")
+                        .pos,
+                ))
+            } else if let Some(node) = mult.values().next() {
+                parse_operation_definition(node)
+            } else {
+                Err(ParseError::NodeNotFound)
+            }
```

Comment:
Can we actually trigger this? If not, should it be a panic branch?

I'd love a test case that tries to hit it if you can come up with one. It can be in the next PR.

Reply:
Yeah I ended up putting in new error type for it but I can definitely make a `panic` then write a test case for it. That shouldn't be a problem what so ever!